composition-js/src/compose.ts

```diff
@@ -31,7 +32,18 @@ export interface CompositionSuccess {
 }
 
 export function compose(subgraphs: Subgraphs): CompositionResult {
-  const mergeResult = mergeSubgraphs(subgraphs);
+  const upgradeResult = upgradeSubgraphsIfNecessary(subgraphs);
+  if (upgradeResult.errors) {
+    return { errors: upgradeResult.errors };
+  }
+
+  const toMerge = upgradeResult.upgraded;
+  const validationErrors = toMerge.validate();
```

Comment:
We previously assumed the input had already been validated, essentially because `Subgraphs`/`Subraph` was validating by default. But with this PR, we start accepting fed1 subgraph and more importantly some subgraph that are initially "invalid" but that are fixed by `SchemaUpgrader`, so validation happens later. I suppose there is other place this could have been put but it also felt reasonable to have it here to make it extra clear we do validate before composition (especially since re-validating a validated schema (that hasn't changed since its validation) is a no-op, so even if validation happens twice on some paths, it's not a big deal).